pkg/util/hlc/hlc.go

```diff
@@ -143,6 +331,37 @@ func (c *Clock) getPhysicalClockLocked() int64 {
 			c.mu.monotonicityErrorsCount++
 			log.Warningf(context.TODO(), "backward time jump detected (%f seconds)", float64(-interval)/1e9)
 		}
+
+		// Physical time should not cross the persisted future wall time (if futureWallTime is set)
+		if c.mu.futureWallTime != 0 && c.mu.lastPhysicalTime > c.mu.futureWallTime {
+			log.Fatalf(
+				context.TODO(),
+				"Cluster setting server.hlc.persist_future_wall_time does not allow physical time "+
+					"%d to be greater than persisted wall time %d.",
+				c.mu.lastPhysicalTime,
+				c.mu.futureWallTime,
+			)
+		}
+
+		toleratedJump := int64(ToleratedOffset(c.mu.maxClockJump))
+		if toleratedJump > 0 {
+			if interval > int64(toleratedJump/10) {
+				log.Fatalf(
+					context.TODO(),
+					"detected backward time jump of %f seconds is not allowed with tolerance of %f seconds",
+					float64(interval)/1e9,
+					float64(c.mu.maxClockJump)/1e9,
+				)
+			}
+			if interval < -toleratedJump {
```

Comment:
I don't see the point of this check. If you crash the process after detecting a forward jump, it will come back up immediately and proceed without any hiccups with the increased time until the time jumps back.

Reply:
It will be useful if we decide to do a force NTP sync on server start-up. So a clock jump => immediate crash => force NTP sync

Reply:
https://github.com/cockroachdb/cockroach/issues/23214

Reply:
If we are making this change dependent on the assumption that cockroach restarts will be accompanied by force ntp syncs, then I would like to see experimental evidence that force NTP syncs fix such jumps, and don't make the situation worse for other users of time before we make this change or start using it.

Reply:
This change will prevent one node of CRDB from using a "jumped" future timestamp in any operation (the other nodes in a cluster could still be using the correct time) before crashing itself due to clock skew. cockroachdb#23214

Reply:
I would say lets not make any assumptions about force NTP syncing on restarts for the purpose of building this because we need more experimental evidence to say that it works as we think it works.

Given that, for multi node deployments, crashing after a forward jump can be useful in some cases because on a restart cockroach will try to detect clock skew and if a node jumped in isolation there is a good chance the skew is detected. If this is the behavior we are relying on, we must verify it using an end to end test that it works as we expect. 

Also this argument does not apply to single node deployments, so we may choose to not turn this on for single nodes.

Reply:
> If this is the behavior we are relying on, we must verify it using an end to end test that it works as we expect.

Yes, I'll write an E2E test after an initial review by CRDB folks. 

> Also this argument does not apply to single node deployments, so we may choose to not turn this on for single nodes.

Correct